server/plugins/configuration/configuration.go

```diff
@@ -0,0 +1,127 @@
+package configuration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/99designs/httpsignatures-go"
+
+	"github.com/woodpecker-ci/woodpecker/server/model"
+	"github.com/woodpecker-ci/woodpecker/server/remote"
+)
+
+type ConfigAPI struct {
+	endpoint string
+	secret   string
+}
+
+// Same as remote.FileMeta but with json tags and string data
+type config struct {
+	Name string `json:"name"`
+	Data string `json:"data"`
+}
+
+type requestStructure struct {
+	Repo          *model.Repo  `json:"repo"`
+	Build         *model.Build `json:"build"`
+	Configuration []*config    `json:"config"`
+}
+
+func NewAPI(endpoint, secret string) ConfigAPI {
+	return ConfigAPI{endpoint: endpoint, secret: secret}
+}
+
+func (cp *ConfigAPI) IsConfigured() bool {
+	return cp.endpoint != ""
+}
+
+func (cp *ConfigAPI) FetchExternalConfig(ctx context.Context, repo *model.Repo, build *model.Build, currentConfig []*remote.FileMeta) (configData []*remote.FileMeta, useOld bool, err error) {
+	// Create request, needs repo
+	response := struct {
+		Pipelines []config `json:"pipelines"`
+	}{}
+
+	currentYamls := make([]*config, len(currentConfig))
+	for i, pipe := range currentConfig {
+		currentYamls[i] = &config{Name: pipe.Name, Data: string(pipe.Data)}
+	}
+
+	status, err := sendRequest(ctx, "POST", cp.endpoint, cp.secret, requestStructure{Repo: repo, Build: build, Configuration: currentYamls}, &response)
```

Comment:
Could we change `sendRequest` to not be call by ref on `&response`.

Reply:
Hm, I took the function from some other part of the codebase, it is mainly done so that the json decoder does use the correct type.... I ca make it explicit defined if you want

Reply:
changed